README.md

```diff
@@ -1,2 +1,46 @@
 # action-s3-artifact
-GitHub action for upload and deploy S3 artifacts
+GitHub action to upload and deploy S3 artifacts
+
+# Usage
+
+See [action.yml](action.yml).
+
+Upload:
+```yaml
+steps:
+  - name: Upload
+    uses: moneymeets/action-s3-artifact@master
+    with:
+      cmd: upload
+      config: s3-artifact-config.yml
+      artifacts_s3_path: s3://YourBucketHere/${{ github.event.repository.name }}/${{ github.sha }}/
+```
+
+Deploy
+```yaml
+  - name: Deploy
+    uses: moneymeets/action-s3-artifact@master
+    with:
+      cmd: deploy
+      config: s3-artifact-config.yml
+      artifacts_s3_path: s3://YourBucketHere/${{ github.event.repository.name }}/${{ github.sha }}/
+      environment: ${{ github.event.deployment.environment }}
+```
+
+Config example:
+
+```yaml
+# s3-artifact-config.yml
+
+target_buckets:
+  dev: dummy.dev
+  live: dummy.live
```

Comment:
Yeah it is not optimal, but I think it is okay for now. GitHub actions doesn't support repository variables at the moment, so it is possible with secrets only. We can set the JSON string as a secret, but then you can't see the correct target bucket in the logfile. I think this is more important.

With Pulumi we can set only secret values. Sure we can put it into secrets, but it is definitely not a secret.

With BitBucket it was possible to set repository variables and Secrets, so looking forward to the feature on GitHub. :)

Reply:
If you want to print the secrets into the logs just put a space between each character or even use dirty tricks like adding an invisible unicode space character between every letter of the secret :)

Reply:
Sounds like a bad hack and I'm not sure if this is readable in our logfile anyways. So in my opinion we need to find a general solution for this, because we have more usecases. (e.g. buckets and secret arn for our mail-dispatcher, etc...)